core/src/main/java/org/apache/iceberg/variants/VariantMetadata.java

```diff
@@ -34,4 +34,20 @@ public interface VariantMetadata extends Variants.Serialized {
 
   /** Returns the size of the metadata dictionary. */
   int dictionarySize();
+
+  static String asString(VariantMetadata metadata) {
+    StringBuilder builder = new StringBuilder();
+
+    builder.append("VariantMetadata(dict={");
+    for (int i = 0; i < metadata.dictionarySize(); i += 1) {
+      if (i > 0) {
+        builder.append(", ");
+      }
+
+      builder.append(i).append(" => ").append(metadata.get(i));
```

Comment:
Nit: Why not just use ":" here as well?

Reply:
+1. Basically we are creating an ID => object key string mapping. Seems fine to use ":" to be consistent.

Reply:
It's not an object, so this isn't really a key. It's a mapping from ID.

Reply:
Sorry. I mean the the strings in the dictionary are the keys from the variant object.